RELEASE-NOTES.md

```diff
@@ -11,6 +11,7 @@
   * `snow notebook create` proving an option to create a Snowflake Notebook from a file on stage.
 
 ## Fixes and improvements
+* `snow app deploy` now scans specified directories for changes even if there is no artifact rule matching that directory
```

Comment:
Any thoughts on how to make that clearer, but not too low level? I'm not super happy with this phrasing, but that's the best I got ATM.

Reply:
Perhaps...
```suggestion
* Passing a directory to `snow app deploy` will now deploy any of its subfolders specified in the application's artifact rules
```